lib/OpenQA/Shared/GruJob.pm

```diff
@@ -10,6 +10,12 @@ sub execute {
     my $self = shift;
 
     my $gru_id = $self->info->{notes}{gru_id};
+    if ($gru_id and not $self->info->{finished}) {
+        # We have a gru_id and this is the first run of the job
+        my $gru = $self->minion->app->schema->resultset('GruTasks')->find($gru_id);
+        # GruTask might not yet have landed in database due to open transaction
+        return $self->retry({delay => 2}) unless $gru;
+    }
```

Comment:
What `retry` exactly does here? does it cancel current scheduled job and create a new one?

Reply:
It aborts the current minion job and schedules to run the same minion  job again after 2 seconds.

Reply:
And IIUC jobs history will ends with those canceled jobs. I just wonder how much acceptable is this from the qa perspective. Maybe not so serious impact but I dont really know how to avoid it.

Reply:
thanks @perlpunk for explanation.resolved

Reply:
Discussed in chat:
We are talking about **minion** jobs here (not openQA jobs), and those are also not cancelled, but retried, and that will run the very same minion job with the same id, and the `retries` field will be incremented.

The related openQA job will wait in the `scheduled` state until the minion job has run successfully and deleted the entry in the GruTasks table. Only then it will be assigned and start running at some point.